nanoFramework.Hardware.Esp32.Rmt/ReceiverChannel.cs

```diff
@@ -13,79 +13,79 @@ namespace nanoFramework.Hardware.Esp32.Rmt
     /// </summary>
     public class ReceiverChannel : RmtChannel, IDisposable
     {
-        #region Fields
-
         [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
-        private TimeSpan _receiveTimeout = new TimeSpan(0, 0, 1); // 1 sec
+        private readonly ReceiverChannelSettings _receiverChannelSettings;
 
-        #endregion Fields
+        /// <inheritdoc/>
+        public override ChannelMode Mode => ChannelMode.Receive;
 
         /// <summary>
-        /// The receive time-out used when calling GetAllItems. Default 1 second.
+        /// Gets or sets the idle threshold after which the receiver will go into idle mode 
+        /// and <see cref="RmtCommand"/>s are copied into the ring buffer and availble to your code.
         /// </summary>
-        public TimeSpan ReceiveTimeout
+        public ushort IdleThreshold
         {
-            get => _receiveTimeout;
-            set => _receiveTimeout = value;
+            get => _receiverChannelSettings.IdleThreshold;
+            set
+            {
+                NativeRxSetIdleThresold(value);
+                _receiverChannelSettings.IdleThreshold = value;
+            }
         }
+
         /// <summary>
-        /// Public constructor to create receiver channel object.
+        /// Gets or sets a value indicating if the the filter is enabled. 
+        /// If enabled, the receiver will ignore pulses with widths less than specified in <see cref="FilterThreshold"/>.
         /// </summary>
-        /// <param name="gpio">The GPIO pin number that we want to use for receiving</param>
-        /// <param name="rmtBufferSize">The maximum number of RMT commands to be reserved in receiver input buffer. 
-        /// Default 100.</param>
-        public ReceiverChannel(int gpio, int rmtBufferSize = 100)
+        public bool EnableFilter
         {
-            // Default to 1MHZ tick ( 1us )
-            _clockDivider = 80;
-
-            // Initialise channel with GPIO and ring buffer size
-            _channel = NativeRxInit(gpio, rmtBufferSize);
+            get => _receiverChannelSettings.EnableFilter;
+            set
+            {
+                NativeRxEnableFilter(value, _receiverChannelSettings.FilterThreshold);
+                _receiverChannelSettings.EnableFilter = value;
+            }
         }
 
         /// <summary>
-        /// Start receiving data on channel.
+        /// Gets or sets the threshold for pulse widths to ignore when <see cref="EnableFilter"/> is set to <see langword="true"/>.
```

Comment:
The comment should offer a clearer explanation about what exactly `width` here will be translated to. The developer, if not familiar with the IDF API, will ask: is this ms? Pulse counts? What? 🤪

Reply:
Agreed and great point! I will update this one ASAP.